File: app/src/main/java/com/majeur/psclient/model/TeamPokemon.java
package com.majeur.psclient.model;

import java.util.Arrays;

public class TeamPokemon extends BasePokemon {

    public static TeamPokemon dummyPokemon() {
        return new TeamPokemon("MissingNo");
    }

    public String name;
    public String item;
    public String ability;
    public String[] moves;
    public String nature;
    public Stats evs;
    public String gender;
    public Stats ivs;
    public boolean shiny;
    public int level;
    public int happiness;
    public String hpType;
    public String pokeball;

    public TeamPokemon(String species) {
        super(species);
        ivs = new Stats(31);
        evs = new Stats(0);
        level = 100;
        happiness = 255;
    }

    @Override
    public String toString() {
        return "Poke{" +
                "name='" + name + '\'' +
                ", species='" + species + '\'' +
                ", item='" + item + '\'' +
                ", ability='" + ability + '\'' +
                ", moves=" + Arrays.toString(moves) +
                ", nature='" + nature + '\'' +
                ", evs=" + evs +
                ", gender='" + gender + '\'' +
                ", ivs=" + ivs +
                ", shiny=" + shiny +
                ", level=" + level +
                ", happiness=" + happiness +
                ", hpType='" + hpType + '\'' +
                ", pokeball='" + pokeball + '\'' +
                '}';
    }
}

File: app/src/main/java/com/majeur/psclient/model/pokemon/TeamPokemon.kt
package com.majeur.psclient.model.pokemon

import com.majeur.psclient.model.common.Stats

class TeamPokemon(species: String) : BasePokemon() {

    var name: String? = null
    var item: String? = null
    var ability: String? = null
    var moves = mutableListOf<String>()
    var nature: String? = null
    var evs: Stats
    var gender: String? = null
    var ivs: Stats
    var shiny = false
    var level: Int
    var happiness: Int
    var hpType: String? = null
    var pokeball: String? = null

    init {
        this.species = species
        ivs = Stats(31)
        evs = Stats(0)
        level = 100
        happiness = 255
    }

    companion object {

        @JvmStatic
        fun dummyPokemon(): TeamPokemon {
            return TeamPokemon("MissingNo")
        }

    }
}
